src/create-with-edge-spec.ts

```diff
@@ -1,23 +1,125 @@
-import { SetupParams } from "./types/setup-params.js"
-import { EdgeSpecRouteFn } from "./types/web-handler.js"
-
-export const createWithEdgeSpec = (globalSpec: SetupParams) => {
-  return (routeSpec: any) =>
-    (routeFn: EdgeSpecRouteFn): EdgeSpecRouteFn =>
-    async (req) => {
-      // Identify environment this is being executed in and convert to WinterCG-
-      // compatible request
-
-      // Execute middleware + auth middleware etc.
-      for (const auth_method of routeSpec.auth ?? []) {
-        // ...
+import type { GlobalSpec } from "./types/global-spec.js"
+import { Middleware } from "./types/middleware.js"
+import { CreateWithRouteSpecFn } from "./types/route-spec.js"
+import {
+  EdgeSpecRequest,
+  EdgeSpecRouteFn,
+  setEdgeSpecRequestOptions,
+} from "./types/web-handler.js"
+
+export const createWithEdgeSpec = <const GS extends GlobalSpec>(
+  globalSpec: GS
+): CreateWithRouteSpecFn<GS> => {
+  return (routeSpec) =>
+    (routeFn): EdgeSpecRouteFn =>
+    async (request) => {
+      const handleErr = async (e: any) => {
+        const exceptionRouteFn = globalSpec.exceptionHandlingMiddleware?.(e)
+
+        if (exceptionRouteFn) {
+          return await exceptionRouteFn(request)
+        }
+
+        throw e
+      }
```

Comment:
imo this probably doesn't need a separate option, can people just add a middleware instead?

Reply:
It isn't really a middleware, it's actually a separate route function. I think `exceptionHandlingRoute` would be more accurate, no?

Reply:
yep, didn't fully understand the middleware system when I made this comment